packages/types/src/sdk/events/app.ts

```diff
@@ -3,50 +3,61 @@ import { BaseEvent } from "./event"
 export interface AppCreatedEvent extends BaseEvent {
   appId: string
   version: string
+  name: string
```

Comment:
Similar case as emails re: PII. Doing a blanket `delete properties.name` feels more risky though. Do we need the name here in the event at all, given we're already enriching the whole app into the payload on read?

Reply:
Same for all the `name` additions to the other events below. We're getting into risky territory by re-using all the posthog property payload types for the audit logs. I get the intention with wanting to put real data into the audit event, it gives a more accurate view of history and prevents you needing to enrich things on read, particularly with things that have since been deleted. 

We might want to consider splitting the event types, e.g. 
```ts
// this can contain as much info as we want, can be written for audit
interface AppCreatedEvent extends BaseEvent {} 

// we must map specifically to this event in the posthog processor
interface AppCreatedAnalyticsEvent extends BaseEvent, AnalyticsEvent {}
```

e.g. in the posthog processor we can add a new integration for getting the analytics representation of an event
```ts

import * as mapping from "./mapping"

async processEvent(
    event: Event,
    identity: Identity,
    properties: BaseEvent,
    timestamp?: string | number
  ): Promise<void> {

  const properties = mapping.mapEvent(event, properties)
}

// ---- mapping.ts ---

const MAPPERS: { [key in Event]: () => AnalyticsEvent } = {
   [Event.APP_CREATED]: mapAppAnalyticsEvent
}

export function mapEvent(event: Event, properties: BaseEvent) {
   return MAPPERS[event](properties)
}

// APPS

function mapAppAnalyticsEvent(event: AppCreatedEvent): AppCreatedAnalyticsEvent {
   return {
      // pull out the specific fields required 
   }
}
```